src/separations.cc

```diff
@@ -150,26 +150,32 @@ std::set<cyclus::RequestPortfolio<Material>::Ptr>
 Separations::GetMatlRequests() {
   using cyclus::RequestPortfolio;
   std::set<RequestPortfolio<Material>::Ptr> ports;
-  bool exclusive = false;
 
-  if (feed.space() > cyclus::eps()) {
-    RequestPortfolio<Material>::Ptr port(new RequestPortfolio<Material>());
+  int t = context()->time();
+  int t_exit = exit_time();
+  if (t_exit >= 0 && (feed.quantity() >= (t_exit - t) * throughput)) {
+    return ports; // already have enough feed for remainder of life
+  } else if (feed.space() < cyclus::eps()) {
+    return ports;
+  }
 
-    Material::Ptr m = cyclus::NewBlankMaterial(feed.space());
-    if (!feed_recipe.empty()) {
-      Composition::Ptr c = context()->GetRecipe(feed_recipe);
-      m = Material::CreateUntracked(feed.space(), c);
-    }
+  bool exclusive = false;
+  RequestPortfolio<Material>::Ptr port(new RequestPortfolio<Material>());
 
-    std::vector<cyclus::Request<Material>*> reqs;
-    for (int i = 0; i < feed_commods.size(); i++) {
-      std::string commod = feed_commods[i];
-      double pref = feed_commod_prefs[i];
-      reqs.push_back(port->AddRequest(m, this, commod, pref, exclusive));
-    }
-    port->AddMutualReqs(reqs);
-    ports.insert(port);
+  Material::Ptr m = cyclus::NewBlankMaterial(feed.space());
+  if (!feed_recipe.empty()) {
+    Composition::Ptr c = context()->GetRecipe(feed_recipe);
+    m = Material::CreateUntracked(feed.space(), c);
+  }
+
+  std::vector<cyclus::Request<Material>*> reqs;
+  for (int i = 0; i < feed_commods.size(); i++) {
+    std::string commod = feed_commods[i];
+    double pref = feed_commod_prefs[i];
```

Comment:
It is more readable as is.  And http://stackoverflow.com/questions/982963/is-there-any-overhead-to-declaring-a-variable-within-a-loop-c